src/Api/InstagramApi.php

```diff
@@ -18,26 +19,34 @@ final class InstagramApi extends SocialApi
     private $config;
 
     /**
-     * @param array $oAuthConfig
+     *  get ssl config
      */
-    public function __construct($oAuthConfig)
+    public function __construct()
     {
-        $this->api = new Instagram();
-        $this->api->setClientID($oAuthConfig['client_id']);
-        $this->api->setAccessToken($oAuthConfig['access_token']);
-
-        $this->config = $oAuthConfig;
+        $grav = Grav::instance();
+        $config = $grav['config']->get('plugins.social-feed');
+        $this->config['enablessl'] = $config['enablessl'];
+        $this->config['certpath'] = $config['certpath'];
     }
 
     /**
      * {@inherit}.
      */
     public function getUserPosts($feed)
     {
-        $user = $this->api->getCurrentUser($feed['username']);
-        $media = $user->getMedia();
 
-        return $media->getData();
+        //save user accesstoken if is set
+        if(isset($feed['userid']) && !empty($feed['userid'])) {
+            $this->config['userid'] = $feed['userid'];
+            $this->config['access_token'] = "&access_token=".$feed['access_token'];
+            $this->config['avatar'] = array_key_first($feed['avatar']);
```

Comment:
@Thoomyy Could you please replace array_key_first by something compatible with php 5.6?
You could use something like:
```php
$this->config['avatar'] = is_array($feed['avatar']) ? key(array_slice($feed['avatar'], 0, 1)) : null;
```

Reply:
Why PHP 5.6?
PHP 5.6 ist End of Life